analysis/feasibility.R

```diff
@@ -0,0 +1,121 @@
+# This script does the following:
+# 1. Assess key feasibility indicators re DM diagnoses, Metformin prescription and COVID diagnoses
+################################################################################
+
+################################################################################
+# 0.0 Import libraries + functions
+################################################################################
+library('arrow')
+library('readr')
+library('here')
+library('lubridate')
+library('dplyr')
+library('tidyr')
+library('purrr')
+library('forcats')
+library('ggplot2')
+library('gt')
+## Import custom user functions
+source(here::here("analysis", "functions", "fn_extract_data.R"))
+source(here::here("analysis", "functions", "fn_case_when.R"))
+source(here::here("analysis", "functions", "fn_diabetes_algorithm.R"))
+
+################################################################################
+# 0.1 Create directories for output
+################################################################################
+fs::dir_create(here::here("output", "data"))
+fs::dir_create(here::here("output", "data_properties"))
+
+################################################################################
+# 1 Import data
+################################################################################
+data_processed <- read_rds(here::here("output", "data", "data_processed.rds"))
+
+################################################################################
+# 2 Check feasibility
+################################################################################
+## 1. Raw incidence curves for the different diabetes diagnoses and PCOS
+# Using the diabetes algo variables for T2DM, T1DM and GestDM
+data_processed <- data_processed %>% 
+  mutate(
+    year_t2dm = year(ymd(cov_date_t2dm)),  
+    year_t1dm = year(ymd(cov_date_t1dm)),
+    year_gestationaldm = year(ymd(cov_date_gestationaldm)),
+    year_pcos = year(ymd(cov_date_pcos)),
+    year_predm = year(ymd(cov_date_prediabetes))
+  ) %>%
+  pivot_longer(
+    cols = starts_with("year_"),       
+    names_to = "diagnosis",                  
+    values_to = "year",              
+    names_prefix = "year_"
+  ) %>%
+  filter(!is.na(year))                
+# Summarize incidence by year and diagnosis
+incidence_data <- data_processed %>%
+  group_by(diagnosis, year) %>%
+  summarise(count = n(), .groups = 'drop')
+# Plot the incidence counts
+plot_incidence <- ggplot(incidence_data, aes(x = year, y = count, color = diagnosis)) +
+  geom_line(size = 1.2) +                   
+  geom_point(size = 3) +  
+  labs(title = "Raw incidence counts of diagnoses over time",
+       x = "Year",
+       y = "Number of Cases") +
+  scale_color_manual(values = c("t2dm" = "blue", "t1dm" = "red", "gestationaldm" = "green", "pcos" = "yellow", "predm" = "orange")) + 
+  theme_minimal()
+
+## 2. How many with a T2DM diagnosis start metformin and median time to start ?
+# T2DM variable from diabetes algo
```

Comment:
Related to above, you probably want to explore more than just 10 days to confirm whether 10 days is the correct choice.

Reply:
I agree, added a few more exploratory thresholds. In my understanding these are two different questions. 

"How many with a T2DM diagnosis start metformin" aims to explore the feasibility for "what is the effect of metformin on the risk of getting sars-cov-2 (and subsequently long covid)". This does not need a grace period but we would probably use monthly sequential trials -> added monthly intervals/periods to explore this. Here, I guess I should already add some outcome feasibility indicators, too, e.g. how many of the metformin starters get covid and how many of those get long covid?

While the other question "How many with a T2DM diagnosis start metformin within 10d after (the first) pos. SARS-CoV-2 test" answers the feasibility for "what is the effect of metformin on the risk of getting severe sars-cov-2 (and subsequently long covid) after getting covid". And here, longer than 10 days makes little sense clinically and lower than 10 days would be a nice to have. But again, fully agree to add more details.